packages/front-end/components/Experiment/Templates/TemplateForm.tsx

```diff
@@ -0,0 +1,360 @@
+              <div className="alert alert-warning">
+                You are creating a template under the demo datasource project.
+                This template will be deleted when the demo datasource project
+                is deleted.
+              </div>
+            )}
+
+            <h4 className="mb-3">Template Details</h4>
+
+            <Field
+              label="Template Name"
+              required
+              minLength={2}
+              {...nameFieldHandlers}
+            />
+
+            {projects.length >= 1 && (
+              <div className="form-group">
+                <label>Available in Project(s)</label>
+                <MultiSelectField
+                  value={form.watch("projects") ?? []}
+                  onChange={(p) => {
+                    form.setValue("projects", p);
+                  }}
+                  name="projects"
+                  initialOption={allowAllProjects ? "All Projects" : undefined}
+                  options={availableProjects}
+                />
+              </div>
+            )}
+
+            <Field
+              label="Template Description"
+              textarea
+              minRows={1}
+              {...form.register("templateMetadata.description")}
+              placeholder={"Short human-readable description of the template"}
+            />
+            <div className="form-group mb-3">
+              <label>Template Tags</label>
+              <TagsInput
+                value={form.watch("templateMetadata.tags") ?? []}
+                onChange={(tags) =>
+                  form.setValue("templateMetadata.tags", tags)
+                }
+              />
+            </div>
+
+            <hr />
+
+            <h4 className="my-3">Experiment Details</h4>
+
+            <Field
+              label="Experiment Hypothesis"
+              textarea
+              minRows={1}
+              placeholder="e.g. Making the signup button bigger will increase clicks and ultimately improve revenue"
+              {...form.register("hypothesis")}
+            />
+
+            <Field
+              label="Experiment Description"
+              textarea
+              minRows={1}
+              {...form.register("description")}
+              placeholder={"Short human-readable description of the experiment"}
+            />
+
+            <div className="form-group">
+              <label>Experiment Tags</label>
+              <TagsInput
+                value={form.watch("tags") ?? []}
+                onChange={(tags) => form.setValue("tags", tags)}
+              />
+            </div>
+          </div>
+        </Page>
+
+        {["Overview", "Traffic", "Targeting", "Metrics"].map((p, i) => {
+          // skip, custom overview page above
+          if (i === 0) return null;
+          return (
+            <Page display={p} key={i}>
+              <ExperimentRefNewFields
+                step={i}
+                source="experiment"
+                project={project}
```

Comment:
Experiment fields use a single project to determine options for certain fields that might change depending on the project, but templates currently can be created for multiple projects. Do we need to make templates limited to one project like we do for experiments for compatibility with `ExperimentRefNewFields`?

Reply:
Right now I'm passing in the current project which is definitely not what we want to do